src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -189,6 +189,84 @@ private string CreateJsonWebToken(JObject payload, SigningCredentials signingCre
                 return rawData;
         }
 
+        /// <summary>
+        /// Decrypts a JWE and returns the clear text 
+        /// </summary>
+        /// <param name="jwtToken">the JWE that contains the cypher text.</param>
+        /// <param name="validationParameters">contains crypto material.</param>
+        /// <returns>the decoded / cleartext contents of the JWE.</returns>
+        /// <exception cref="ArgumentNullException">if 'jwtToken' is null.</exception>
+        /// <exception cref="ArgumentNullException">if 'validationParameters' is null.</exception>
+        /// <exception cref="SecurityTokenException">if 'jwtToken.Enc' is null or empty.</exception>
+        /// <exception cref="SecurityTokenEncryptionKeyNotFoundException">if 'jwtToken.Kid' is not null AND decryption fails.</exception>
+        /// <exception cref="SecurityTokenDecryptionFailedException">if the JWE was not able to be decrypted.</exception>
+        protected string DecryptToken(JsonWebToken jwtToken, TokenValidationParameters validationParameters)
+        {
+            if (jwtToken == null)
+                throw LogHelper.LogArgumentNullException(nameof(jwtToken));
+
+            if (validationParameters == null)
+                throw LogHelper.LogArgumentNullException(nameof(validationParameters));
+
+            if (string.IsNullOrEmpty(jwtToken.Enc))
+                throw LogHelper.LogExceptionMessage(new SecurityTokenException(LogHelper.FormatInvariant(TokenLogMessages.IDX10612)));
+
+            var keys = GetContentEncryptionKeys(jwtToken, validationParameters);
+
+            // keep track of exceptions thrown, keys that were tried
+            StringBuilder exceptionStrings = new StringBuilder();
```

Comment:
use var

Reply:
I went through the new code that was added and switched to 'var' wherever possible.